[SEC-1639] Add impossible travel detection methods (#13204)
DataDog/documentation: content/en/security_platform/cloud_siem/log_detection_rules.md

**Note**: The query applies to all Datadog events and ingested logs that do not require indexing.

{{% /tab %}}

{{% tab "Impossible Travel" %}}

### Search query

Construct a search query using the same logic as a [log explorer search][1]. All logs matching this query are analyzed for a potential impossible travel. You can use the `preview` section to see which logs are matched by the current query.

#### User attribute

> For the `user attribute`, select the field in the analyzed log that contains the user ID. This can be an identifier like an email address, user name, account identifier.

Reviewer: ```suggestion
For the `user attribute`, select the field in the analyzed log that contains the user ID. This can be an identifier like an email address, user name, or account identifier.
```

#### Location attribute

> The `location attribute` specifies which field holds the geographic information for a log. It cannot currently be changed and defaults to `@network.client.geoip`, which is the output of the [GeoIP parser][2] logs processor which enriches a log with location information based on the client's IP address.

Reviewer 2: > It cannot currently be changed and defaults to

How about something like "The only supported value is `@network.client.geoip` which is enriched by the [GeoIP parser][2]`?

> The `location attribute` specifies which field holds the geographic information for a log. It cannot currently be changed and defaults to `@network.client.geoip`, which is the output of the [GeoIP parser][2] logs processor which enriches a log with location information based on the client's IP address.

Reviewer: ```suggestion
The `location attribute` specifies which field holds the geographic information for a log. The only supported value is `@network.client.geoip`, which is enriched by the [GeoIP parser][2] to give a log location information based on the client's IP address.
```

#### Baseline user locations

> Tick this checkbox if you'd like Datadog to learn regular access locations before triggering a signal.

Reviewer: ```suggestion
Click the checkbox if you'd like Datadog to learn regular access locations before triggering a signal.
```

> Tick this checkbox if you'd like Datadog to learn regular access locations before triggering a signal.

Reviewer: ```suggestion
Click the checkbox if you'd like Datadog to learn regular access locations before triggering a signal.
```

> If selected, no signal will be triggered in the first 24 hours. In that time, Datadog learns the user's regular access locations. This can be helpful to reduce noise and infer VPN usage or credentialed API access.

Reviewer 2: ```suggestion
When selected, signal are suppressed for the first 24 hours. In that time, Datadog learns the user's regular access locations. This can be helpful to reduce noise and infer VPN usage or credentialed API access.
```

> Leave the checkbox unticked to detect all impossible travel behavior.

Reviewer: ```suggestion
Do not click the checkbox if you want Datadog to detect all impossible travel behavior.
```

A signal will "close" regardless of whether or not the anomaly is still anomalous once the time exceeds the maximum signal duration. This time is calculated from the first seen timestamp.

[1]: /monitors/notify/?tab=is_alert#integrations
{{% /tab %}}

{{% tab "Impossible Travel" %}}

The impossible travel detection method does not require setting a rule case.

### Severity and notification

Select an appropriate severity level for the security signal (`INFO`, `LOW`, `MEDIUM`, `HIGH`, `CRITICAL`).

In the “Notify” section, configure zero or more [notification targets][1].

### Time windows

> An `evaluation window` is specified to match when at least one of the cases matches true. This is a sliding window and evaluates in real-time.

Reviewer 2: I know this is the same phrasing as for the other rule types but `matches true` seems off to me.

> An `evaluation window` is specified to match when at least one of the cases matches true. This is a sliding window and evaluates in real-time.

Reviewer: ```suggestion
An `evaluation window` is specified to match when at least one of the cases is true. This is a sliding window and evaluates in real-time.
```

> Once a signal is generated, the signal will remain “open” if a case is matched at least once within this `keep alive` window. Each time a new event matches any of the cases, the *last updated* timestamp is updated for the signal.

Reviewer: ```suggestion
Once a signal is generated, the signal remains “open” if a case is matched at least once within the `keep alive` window. Each time a new event matches any of the cases, the *last updated* timestamp is updated for the signal.
```

> A signal will “close” regardless of the query being matched once the time exceeds the `maximum signal duration`. This time is calculated from the first seen timestamp.

Reviewer: ```suggestion
A signal closes regardless of the query being matched once the time exceeds the `maximum signal duration`. This time is calculated from the first seen timestamp.
```
